### PR TITLE
feat: add mav2 account

### PR DESCRIPTION
**Pull request by howydev**
uses a placeholder signer for now

---

## PR-Codex overview
This PR introduces the `SMAV2Account` type and a function `createSMAV2Account` to facilitate the creation of smart contract accounts with enhanced parameters for transport, signer, and entry point management.

### Detailed summary
- Added `SMAV2Account` type definition.
- Created `CreateSMAV2AccountParams` type for account creation parameters.
- Implemented `createSMAV2Account` function for initializing smart contract accounts.
- Included logic to handle account initialization and address retrieval.
- Integrated `standardExecutor` and `multiOwnerMessageSigner` for enhanced functionality.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by linnall — APPROVED** [review]

File: account-kit/smart-contracts/src/ma-v2/account/account.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import { standardExecutor } from "../../msca/account/standardExecutor.js";
+import { multiOwnerMessageSigner } from "../../msca/plugins/multi-owner/signer.js"; // TODO: swap for MA v2 signer
+
+export const DEFAULT_OWNER_ENTITY_ID = 0;
+
+export type SMAV2Account<
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = SmartContractAccountWithSigner<"SMAV2Account", TSigner, "0.7.0">;
+
+export type CreateSMAV2AccountAccountParams<
```
Comment (linnall) [inline account-kit/smart-contracts/src/ma-v2/account/account.ts:32]: did you mean `CreateSMAV2AccountParams` here?

Comment (howydev) [reply account-kit/smart-contracts/src/ma-v2/account/account.ts:32]: good catch!

**Comment by howydev** [comment]
### Merge activity

* **Dec 13, 11:12 AM EST**: The merge label 'graphite-merge-queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=alchemyplatform&repo=aa-sdk) once it meets the requirements.

**Comment by howydev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1208?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1209** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1209?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1208** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1208?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1208?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1207** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1207?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1203** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1203?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite-merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/alchemyplatform?ref=merge-queue-instructions-in-stack-comment&prId=1208).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Pull request merged**